portfolio/src/main/webapp/script.js

```diff
@@ -50,3 +51,9 @@ function toggleImages() {
     image.classList.toggle('hide');
   });
 }
+
+async function getMessage() {
+  const response = await fetch('/data');
+  const message = await response.text();
+  document.querySelector('#greeting').innerHTML = message;
```

Comment:
just a small comment. since the returned response is text, it may be better to use innerText instead of innerHTML, as innerHTML is usually not preferred as it could make the website vulnerable to some security problem like xss

Reply:
Ok noted! Used innerHTML this time because innerText added an extra <br>, but will try using innerText if possible.